### PR TITLE
Release automation

### LINKED ISSUE #221: Write some release automation
We should lift some of our release automation from cryptography and bcrypt to make releases easier here.

### PR DESCRIPTION
**Pull request by reaperhulk**
Fixes #221

### CODE REVIEW

**Comment by alex** [comment]
I observe that this is basically a copy-paste of the `pyca/cryptography` automatino.

**Comment by reaperhulk** [comment]
Mostly. I fixed a few small things though so I will be sending a PR to improve cryptography's automation later today based on what I did here.

File: tasks.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    invoke.run("twine upload -s dist/PyNaCl-{0}* ".format(version))
+
+    session = requests.Session()
+
+    # This tells the CDN to delete the cached response for the URL. We do this
+    # so that the Jenkins builders will see the new sdist immediately when they
+    # go to build the wheels.
+    response = session.request(
+        "PURGE", "https://pypi.python.org/simple/pynacl/"
```
Comment (dstufft) [inline tasks.py]: This shouldn't be required if you use Warehouse fwiw. The purging there has fancy things like retries! However it doesn't really hurt to keep it either.

File: tasks.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        "{0}/build".format(JENKINS_URL),
+        params={
+            "cause": "Building wheels for {0}".format(version),
+            "token": token
+        }
+    )
+    response.raise_for_status()
+    wait_for_build_completed(session)
+    paths = download_artifacts(session)
+    invoke.run("twine upload {0}".format(" ".join(paths)))
```
Comment (dstufft) [inline tasks.py]: Do we purposely only sign sdists?

Comment (reaperhulk) [inline tasks.py]: That's how we've done it on cryptography. I don't have any objection to signing more, but I also don't want to type my GPG key password in 20 times :)

Comment (dstufft) [inline tasks.py]: Fine with me, just noticed it.

**Pull request merged**